finalProject/tenSpeckledFrogs.js

```diff
@@ -0,0 +1,143 @@
+function frogs(numInitialFrogs) {
+  var rhyme = '';
+  var firstTime = 1;
+  var lastTime = numInitialFrogs; // when building loop, start i at 1
+  var currNumFrogs = numInitialFrogs;
+  for (var i = 1; i <= numInitialFrogs; i++) {
+    var currMsg = '';
+    if (i === 1) { // build first iteration msg
+      var currNum = numToString(numInitialFrogs);
+      var currNumCap = currNum.charAt(0).toUpperCase() + currNum.slice(1);
+      currMsg = `${currNumCap} speckled frogs sat on a log eating some most delicious bugs.\nOne jumped in the pool where it's nice and cool, then there were ${numToString(numInitialFrogs - 1)} speckled frogs.\n\n`;
+      rhyme += currMsg;
+    } else if (i === numInitialFrogs) { // build last iteration msg
+      currMsg = `One speckled frog sat on a log eating some most delicious bugs.\nOne jumped in the pool where it's nice and cool,\nthen there were no more speckled frogs!`;
+      rhyme += currMsg;
+    } else if (currNumFrogs > 2) { // any other inner msg
+      var currNum = numToString(currNumFrogs);
+      var currNumCap = currNum.charAt(0).toUpperCase() + currNum.slice(1);
+      currMsg = `${currNumCap} speckled frogs sat on a log eating some most delicious bugs.\nOne jumped in the pool where it's nice and cool,\nthen there was ${numToString(currNumFrogs - 1)} speckled frogs.\n\n`;
+      rhyme += currMsg;
+    } else if (currNumFrogs === 2) { // if frogs is two, need specific formatting for one frog left on last line
+      currMsg = `Two speckled frogs sat on a log eating some most delicious bugs.\nOne jumped in the pool where it's nice and cool,\nthen there was one speckled frog.\n\n`;
+      rhyme += currMsg;
+    }
+    currNumFrogs -= 1;
```

Comment:
When I see this, I would say there's probably an opportunity to use a for loop that starts high and goes to 1